src/Reflection/ReflectionFunctionAbstract.php

```diff
@@ -39,13 +39,13 @@ trait ReflectionFunctionAbstract
      * @var array<non-empty-string, ReflectionParameter>
      * @psalm-allow-private-mutation
      */
-    private array $parameters;
+    protected array $parameters;
 
     /** @psalm-allow-private-mutation */
     private bool $returnsReference;
 
     /** @psalm-allow-private-mutation */
-    private ReflectionNamedType|ReflectionUnionType|ReflectionIntersectionType|null $returnType;
+    protected ReflectionNamedType|ReflectionUnionType|ReflectionIntersectionType|null $returnType;
```

Comment:
Not a fan of using `protected`  (plus a trait), so we may really need to do some changing here in the next versions.